src/interaction/click.ts

```diff
@@ -3,18 +3,23 @@ import { ClickData } from "@clarity-types/interaction";
 import { bind } from "@src/core/event";
 import { schedule } from "@src/core/task";
 import { link, target, track } from "@src/data/target";
+import { iframe } from "@src/layout/dom";
 import encode from "./encode";
 
 export let data: ClickData;
 
-export function start(): void {
-    bind(document, "click", handler.bind(this, Event.Click), true);
+export function observe(root: Node): void {
+    bind(root, "click", handler.bind(this, Event.Click, root), true);
 }
 
-function handler(event: Event, evt: MouseEvent): void {
-    let de = document.documentElement;
-    let x = "pageX" in evt ? Math.round(evt.pageX) : ("clientX" in evt ? Math.round(evt["clientX"] + de.scrollLeft) : null);
-    let y = "pageY" in evt ? Math.round(evt.pageY) : ("clientY" in evt ? Math.round(evt["clientY"] + de.scrollTop) : null);
+function handler(event: Event, root: Node, evt: MouseEvent): void {
+    let frame = iframe(root);
+    let d = frame ? frame.contentDocument.documentElement : document.documentElement;
+    let x = "pageX" in evt ? Math.round(evt.pageX) : ("clientX" in evt ? Math.round(evt["clientX"] + d.scrollLeft) : null);
+    let y = "pageY" in evt ? Math.round(evt.pageY) : ("clientY" in evt ? Math.round(evt["clientY"] + d.scrollTop) : null);
+    x = x && frame ? x + frame.offsetLeft : x;
```

Comment:
Minor; Add coment about why need to offset coordinates by iframe top left here.

Reply:
Thanks done.